core/trino-main/src/main/java/io/trino/operator/Distribution.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.airlift.stats.TDigest;
+import io.trino.plugin.base.metrics.TDigestHistogram;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class Distribution
+{
+    @Nullable
+    @JsonSerialize(converter = TDigestHistogram.TDigestToBase64Converter.class)
+    @JsonDeserialize(converter = TDigestHistogram.Base64ToTDigestConverter.class)
+    private final TDigest digest;
+    private final double count;
+    private final double min;
+    private final double max;
+    private final double p01;
+    private final double p05;
+    private final double p10;
+    private final double p25;
+    private final double p50;
+    private final double p75;
+    private final double p90;
+    private final double p95;
+    private final double p99;
+
+    public static Distribution withCount(long count)
```

Comment:
`with...` suggests it is an instance method which adds `count` information to the instance. Maybe name it `ofCount`